packages/shared/src/components/squads/settings/SquadPrivacySection.tsx

```diff
@@ -0,0 +1,132 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+import React, { ReactElement, useEffect, useMemo, useState } from 'react';
+import { gqlClient } from '../../../graphql/common';
+import {
+  SourceCategoryData,
+  SOURCE_CATEGORIES_QUERY,
+} from '../../../graphql/sources';
+import {
+  generateQueryKey,
+  RequestKey,
+  flattenInfiniteQuery,
+} from '../../../lib/query';
+import { Dropdown } from '../../fields/Dropdown';
+import { Radio } from '../../fields/Radio';
+import { StatusDescription } from './common';
+import { SquadSettingsSection } from './SquadSettingsSection';
+
+interface CategoryDropdownProps {
+  initialCategory: string;
+  isPublic: boolean;
+  categoryHint?: string;
+  onCategoryChange?: (category: string) => void;
+}
+
+const classes = {
+  wrapper: 'w-full',
+  content: 'w-fit',
+};
+
+export enum PrivacyOption {
+  Private = 'private',
+  Public = 'public',
+}
+
+export function SquadPrivacySection({
+  initialCategory,
+  isPublic = true,
+  categoryHint,
+  onCategoryChange,
+}: CategoryDropdownProps): ReactElement {
+  const [privacy, setPrivacy] = useState(
+    isPublic ? PrivacyOption.Public : PrivacyOption.Private,
+  );
+  const isPrivate = privacy === PrivacyOption.Private;
+  const [selected, setSelected] = useState(-1);
+  const { data } = useInfiniteQuery<SourceCategoryData>(
```

Comment:
this can 100% use long staleTime actually, don't expect them to change often.

Reply:
Yeah, we can add stale time here.